### PR TITLE
issue #1227 refactor bulkdata jbatch job codes

### PR DESCRIPTION
**Pull request by albertwang-ibm**
Summary of changes:
(1) changed package name to com.ibm.fhir.jbatch.bulkdata.
(2) changed to use the builder pattern for checkpointdata and transientuserdata.
(3) removed unused get/set member functions.

Signed-off-by: Albert Wang <xuwang@us.ibm.com>

### CODE REVIEW

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/common/TransientUserData.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            transientUserData.currentUploadSize = this.currentUploadSize;
+            transientUserData.totalResourcesNum = this.totalResourcesNum;
+            transientUserData.indexOfCurrentTypeFilter = this.indexOfCurrentTypeFilter;
+            transientUserData.resourceTypeSummary = this.resourceTypeSummary;
+            transientUserData.lastWritePageNum = this.lastWritePageNum;
+            return transientUserData;
+        }
+
+    }
+
```
Comment (prb112) [inline fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/common/TransientUserData.java:70]:
```suggestion
    }
```

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/common/TransientUserData.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            .lastPageNum(checkPointData.lastPageNum)
+            .lastWritePageNum(checkPointData.lastWritePageNum)
+            .build();
+    }
+
+    public ByteArrayOutputStream getBufferStream() {
+        return bufferStream;
+    }
+
+
```
Comment (prb112) [inline fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/common/TransientUserData.java:44]:
```suggestion

```

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/group/ChunkReader.java [hunk 3]
```diff
@@ -140,7 +140,21 @@ public Object readItem() throws Exception {
...
+                .indexOfCurrentTypeFilter(0)
+                .resourceTypeSummary(null)
+                .totalResourcesNum(0)
+                .currentUploadResourceNum(0)
+                .currentUploadSize(0)
+                .uploadCount(1)
+                .lastPageNum(0)
+                .lastWritePageNum(1)
+                .build();
+
```
Comment (prb112) [inline fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/group/ChunkReader.java:157]: I really love this code. Very readable. 

Constants would make this more readable, e.g. why null for resourceTypeSummary

Comment (prb112) [reply fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/group/ChunkReader.java:157]: This is not something I'm asking to be recoded/changed.  Just a commentary

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/system/ChunkReader.java [hunk 4]
```diff
@@ -217,8 +217,21 @@ public Object readItem() throws Exception {
...
+                    .partNum(1)
+                    .indexOfCurrentTypeFilter(0)
+                    .resourceTypeSummary(null)
+                    .totalResourcesNum(0)
+                    .currentUploadResourceNum(0)
+                    .currentUploadSize(0)
+                    .uploadCount(1)
+                    .lastPageNum(searchContext.getLastPageNumber())
+                    .lastWritePageNum(1)
+                    .build();
```
Comment (prb112) [inline fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/system/ChunkReader.java:233]: This is duplicate of other ChunkReader implementation, do you want to simplify?

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/load/ImportCheckPointData.java [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+                .partNumForFailureOperationOutcomes(userData.getPartNumForFailureOperationOutcomes())
+                .totalReadMilliSeconds(userData.getTotalReadMilliSeconds())
+                .totalValidationMilliSeconds(userData.getTotalValidationMilliSeconds())
+                .totalWriteMilliSeconds(userData.getTotalWriteMilliSeconds())
+                .importFileSize(userData.getImportFileSize())
+                .inFlyRateBeginMilliSeconds(userData.getInFlyRateBeginMilliSeconds())
+                .build();
+
+
+
```
Comment (prb112) [inline fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/load/ImportCheckPointData.java:270]:
```suggestion
```

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/system/ChunkReader.java (hunk 4 above)
Comment (albertwang-ibm) [reply fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/export/system/ChunkReader.java:233]: Group export reader inherits from Patient reader, but System export reader is very different from them, so I didn't really want to mix them...

**Pull request merged**